docs/source/deployment/deploy-model-to-kubernetes/tutorial.rst

```diff
@@ -0,0 +1,484 @@
+Develop ML model with MLflow and deploy to Kubernetes
+=====================================================
+
+.. note::
+
+  This tutorial assumes that you have access to a Kubernetes cluster. However, you can also complete this tutorial on your local machine
+  by using local cluster emulation tools such as `Kind <https://kind.sigs.k8s.io/docs/user/quick-start>`_ or `Minikube <https://minikube.sigs.k8s.io/docs/start/>`_.
+
+
+This guide demonstrates how to use MLflow end-to-end for:
+
+- Training a linear regression model with `MLflow Tracking <../../../tracking.html>`_.
+- Conducting hyper arameter tuning to find the best model.
+- Packaging the model weight and dependencies as an `MLflow Model <../../../models.html>`_.
+- Testing model serving locally with `mlserver <https://mlserver.readthedocs.io/en/latest/>`_ using the `mlflow models serve <../../../cli.html#mlflow-models-serve>`_ command.
+- Deploying the model to a Kubernetes cluster using `KServe <https://kserve.github.io/website/>`_ with MLflow.
+
+We will cover end-to-end model developemnt process including model training and testing.
+However, if you already have a model and just want to learn how to deploy it to Kubernetes, you can skip to :ref:`Step 6 - Test Model Serving Locally <step-6-test-model-serving-locally>`
+
+
+Introduction: Scalable Model Serving with KServe and MLServer
+-------------------------------------------------------------
+
+MLflow provides an easy-to-use interface for deploying modeles as a Flask-based inference server. You can deploy the same inference
+server to a Kubernetes cluster by containerizing it using the ``mlflow models build-docker`` command. However, this approach may not be scalable
+and could be unsuitable for production use cases. Flask is not designed for high performance, and manually managing multiple instances of
```

Comment:
Can we say why Flask isn't ideally suited for ML inference (it's blocking) and how async gateways are far more optimized due to the potentially long-running nature of inference (depending on the model architecture, size, and optimization of the underlying library) and issues with scalability with a synchronous blocking web gateway?

Reply:
Sure! Will add brief explanation on why.